File: src/test/java/com/github/kayjamlang/tests/expressions/var/VariableTest.java
package com.github.kayjamlang.tests.expressions.var;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.core.expressions.VariableExpression;
import com.github.kayjamlang.tests.TestsUtils;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class VariableTest {

    @Test
    public void test() throws Exception {
        Expression expression = TestsUtils.parse("var test = 123");

        assertNotNull(expression);
        assertSame(VariableExpression.class, expression.getClass());

        VariableExpression variable = (VariableExpression) expression;
        assertEquals("test", variable.name);
        assertSame(ValueExpression.class, variable.expression.getClass());

        ValueExpression constant = (ValueExpression) variable.expression;
        assertEquals(123, constant.value);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/var/VariableTest.scala
package com.github.kayjamlang.tests.expressions.`var`

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.{ValueExpression, VariableExpression}
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object VariableTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("var test = 123"))
  }
}

class VariableTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = VariableTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[VariableExpression], expression getClass)
    val variable = expression.asInstanceOf[VariableExpression]
    assertEquals("test", variable.name)
    assertSame(classOf[ValueExpression], variable.expression getClass)
    val constant = variable.expression.asInstanceOf[ValueExpression]
    assertEquals(123, constant.value)
  }
}
